managed/services/management/backup/artifacts_service_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2022 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package backup
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/brianvoe/gofakeit/v6"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+	"gopkg.in/reform.v1/dialects/postgresql"
+
+	backupv1beta1 "github.com/percona/pmm/api/managementpb/backup"
+	"github.com/percona/pmm/managed/models"
+	"github.com/percona/pmm/managed/services/backup"
+	"github.com/percona/pmm/managed/utils/testdb"
+	"github.com/percona/pmm/managed/utils/tests"
+)
+
+func TestListPitrTimelines(t *testing.T) {
+	ctx := context.Background()
+	sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+	mockedRemovalSvc := &mockRemovalService{}
+	mockedStorageSvc := &mockPitrStorageService{}
+
+	timelines := []backup.Timeline{
+		{
+			ReplicaSet: "rs0",
+			Start:      uint32(time.Now().Unix()),
+			End:        uint32(time.Now().Unix()),
+		},
+	}
+
+	mockedStorageSvc.On("ListPITRTimelines", ctx, mock.Anything, mock.Anything).Return(timelines, nil)
```

Comment:
I suggest to use the expectations functions like [Times](https://pkg.go.dev/github.com/stretchr/testify/mock#Call.Times)
And also check your expectations: [AssertExpectations](https://pkg.go.dev/github.com/stretchr/testify/mock#Mock.AssertExpectations)